Client/Frontend/Browser/SearchEngines.swift

```diff
@@ -140,27 +140,55 @@ class SearchEngines {
         }
     }
 
+    /// Return all possible paths for a language identifier in the order of most specific to least specific.
+    /// For example, zh-Hans-CN with a default of en will return [zh-Hans-CN, zh-CN, zh, en]. The fallback
+    /// identifier must be a known one that is guaranteed to exist in the SearchPlugins directory.
+    class func directoriesForLanguageIdentifier(languageIdentifier: String, basePath: NSString, fallbackIdentifier: String) -> [String] {
+        var directories = [String]()
+        let components = languageIdentifier.componentsSeparatedByString("-")
+        if components.count == 1 {
+            // zh
+            directories.append(languageIdentifier)
```

Comment:
Note that all cases have this line, and it's the first line in each one, so you could simplify this by factoring it out above the `if` statement (and that would allow you to remove the `components.count == 1` case completely).